shared_model/backend/protobuf/queries/impl/proto_tx_pagination_meta.cpp

```diff
@@ -31,7 +34,23 @@ std::optional<types::HashType> TxPaginationMeta::firstTxHash() const {
   }
   return types::HashType::fromHexString(meta_.first_tx_hash());
 }
-
+std::optional<types::TimestampType> TxPaginationMeta::firstTxTime() const {
+  if (meta_.opt_first_tx_time_case()
+      == iroha::protocol::TxPaginationMeta::OptFirstTxTimeCase::
+             OPT_FIRST_TX_TIME_NOT_SET) {
+    return std::nullopt;
+  }
+  std::cout<<"before conversion "<<google::protobuf::util::TimeUtil::TimestampToNanoseconds(meta_.first_tx_time())<<std::endl;
+  return google::protobuf::util::TimeUtil::TimestampToNanoseconds(meta_.first_tx_time());
```

Comment:
`TimestampToMicroseconds` instead of `TimestampToNanoseconds`

Reply:
I think it should actually be TimestampToMilliseconds. Look here : https://iroha.readthedocs.io/en/main/concepts_architecture/glossary.html#transaction

Reply:
You are right, sorry